pkg/batcher/transport_message_batcher.go

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright 2025 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package batcher
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/rs/zerolog/log"
+	"github.com/th2-net/th2-common-go/pkg/queue/message"
+	transport "github.com/th2-net/transport-go/pkg"
+)
+
+type protocol = string
+
+const (
+	th2_pin_attribute string = "transport-group"
+)
+
+type MessageArguments struct {
+	Metadata  map[string]string
+	Alias     string
+	Direction Direction
+	Protocol  protocol
+}
+
+type messageBatcher struct {
+	encoder    transport.Encoder
+	groupIndex int
+
+	pipe        chan transport.RawMessage
+	seqProvider sequenceProvider
+
+	batchSizeBytes int
+	flushTimeout   time.Duration
+
+	router   message.Router
+	book     string
+	group    string
+	protocol protocol
+
+	done chan bool
+}
+
+func NewMessageBatcher(router message.Router, cfg MqMessageBatcherConfig) (MqBatcher[MessageArguments], error) {
+	flushTimeout := cfg.FlushMillis
+	if flushTimeout <= 0 {
+		flushTimeout = DEFAULT_FLUSH_TIME
+	}
+
+	maxBatchSizeBytes := cfg.BatchSizeBytes
+	if maxBatchSizeBytes <= 0 {
+		maxBatchSizeBytes = DEFAULT_BATCH_SIZE
+	}
+
+	flushDuration := time.Duration(flushTimeout) * time.Millisecond
+
+	batcher := messageBatcher{
+		encoder:        transport.NewEncoder(make([]byte, maxBatchSizeBytes)),
+		pipe:           make(chan transport.RawMessage, 1),
```

Comment:
note: Probably, we should make the channel capacity configurable. Right now it will allow only 1 "inflight" message before blocking on sending another message to the channel